packages/blade/src/components/Button/BaseButton/BaseButton.tsx

```diff
@@ -87,12 +114,25 @@ type BaseButtonWithoutIconProps = BaseButtonCommonProps & {
 type BaseButtonWithIconProps = BaseButtonCommonProps & {
   icon: IconComponent;
   children?: StringChildrenType;
+  imgProps?: undefined;
```

Comment:
I'm not sure if this will scale well. We already have a lot of code in BaseButton. Adding extra Avatar related props here might bloat this further. 

Can we instead have a separate `AvatarButton` component which will extend normal html button?

Reply:
Done